packages/flutter_tools/lib/src/build_system/targets/icon_tree_shaker.dart

```diff
@@ -0,0 +1,366 @@
+                   && _environment.defines[kBuildMode] != 'debug';
+
+  /// Fills the [_iconData] map.
+  Future<Map<String, _IconTreeShakerData>> _getIconData(Environment environment) async {
+    if (!enabled) {
+      return null;
+    }
+
+    final File appDill = environment.buildDir.childFile('app.dill');
+    if (!appDill.existsSync()) {
+      throw IconTreeShakerException._('Expected to find kernel file at ${appDill.path}, but no file found.');
+    }
+    final File constFinder = _fs.file(
+      _artifacts.getArtifactPath(Artifact.constFinder),
+    );
+    final File dart = _fs.file(
+      _artifacts.getArtifactPath(Artifact.engineDartBinary),
+    );
+
+    final Map<String, List<int>> iconData = await _findConstants(
+      dart,
+      constFinder,
+      appDill,
+    );
+    final Set<String> familyKeys = iconData.keys.toSet();
+
+    final Map<String, String> fonts = await _parseFontJson(
+      _fontManifest,
+      familyKeys,
+    );
+
+    if (fonts.length != iconData.length) {
+      throwToolExit('Expected to find fonts for ${iconData.keys}, but found '
+                    '${fonts.keys}. This usually means you are refering to '
+                    'font families in an IconData class but not including them '
+                    'in the assets section of your pubspec.yaml, are missing '
+                    'the package that would include them, or are missing '
+                    '"uses-material-design: true".');
+    }
+
+    final Map<String, _IconTreeShakerData> result = <String, _IconTreeShakerData>{};
+    for (final MapEntry<String, String> entry in fonts.entries) {
+      result[entry.value] = _IconTreeShakerData(
+        family: entry.key,
+        relativePath: entry.value,
+        codePoints: iconData[entry.key],
+      );
+    }
+    return result;
+  }
+
+  /// Calls font-subset, which transforms the `inputPath` font file to a
+  /// subsetted version at `outputPath`.
+  ///
+  /// The `relativePath` parameter
+  ///
+  /// All parameters are required.
+  ///
+  /// If [enabled] is false, or the relative path is not recognized as an icon
+  /// font used in the Flutter application, this returns false.
+  /// If the font-subset subprocess fails, it will [throwToolExit].
+  /// Otherwise, it will return true.
+  Future<bool> subsetFont({
+    @required String inputPath,
+    @required String outputPath,
+    @required String relativePath,
+  }) async {
+    if (!enabled) {
+      return false;
+    }
+
+    _iconData ??= await _getIconData(_environment);
+    assert(_iconData != null);
+
+    final _IconTreeShakerData iconTreeShakerData = _iconData[relativePath];
+    if (iconTreeShakerData == null) {
+      return false;
+    }
+
+    final File fontSubset = _fs.file(
+      _artifacts.getArtifactPath(Artifact.fontSubset),
+    );
+    if (!fontSubset.existsSync()) {
+      throwToolExit('The font-subset utility is missing. Run "flutter doctor".');
```

Comment:
This would indicate a programming error in the flutter tool. I think we should just let this one crash if we hit it so we can see it crash logging instead of covering it with a tool exit

Reply:
I was thinking a user could mess this up on us but I guess we should re-download if they do.  Will change.